R/landing_popup_module.R

```diff
@@ -0,0 +1,110 @@
+#' Landing Popup Module
+#'
+#' @description Creates a landing welcome popup for `teal` applications.
+#'
+#' This module is used to display a popup dialog when the application starts.
+#' The dialog blocks the access to the application and must be closed with a button before the application is viewed.
+#'
+#' @param label `character(1)` the label of the module.
+#' @param title `character(1)` the text to be displayed as a title of the popup.
+#' @param content The content of the popup. Passed to `...` of `shiny::modalDialog`. Can be a `character`
+#' or a list of `shiny.tag`s. See examples.
+#' @param buttons `shiny.tag` or a list of tags (`tagList`). Typically a `modalButton` or `actionButton`. See examples.
+#'
+#' @return A `teal_module` (extended with `teal_landing_module` class) to be used in `teal` applications.
+#'
+#' @examples
+#' app1 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       content = "A place for the welcome message or a disclaimer statement.",
+#'       buttons = modalButton("Proceed")
+#'     ),
+#'     example_module()
+#'   )
+#' )
+#' if (interactive()) {
+#'   shinyApp(app1$ui, app1$server)
+#' }
+#'
+#' app2 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       title = "Welcome",
+#'       content = tags$b(
+#'         "A place for the welcome message or a disclaimer statement.",
+#'         style = "color: red;"
+#'       ),
+#'       buttons = tagList(
+#'         modalButton("Proceed"),
+#'         actionButton("close", "Read more",
+#'           onclick = "window.open('http://google.com', '_blank')"
+#'         )
+#'       )
+#'     ),
+#'     example_module()
+#'   )
+#' )
+#'
+#' if (interactive()) {
+#'   shinyApp(app2$ui, app2$server)
+#' }
+#'
+#' app3 <- teal::init(
+#'   data = teal.data::dataset("iris", iris),
+#'   modules = teal::modules(
+#'     teal::landing_popup_module(
+#'       title = "Welcome",
+#'       content = "App will close, once you click the rejection button!",
+#'       buttons = tagList(
+#'         modalButton("Proceed"),
+#'         actionButton("close", "Reject", onclick = "window.close()")
+#'       )
+#'     ),
+#'     module(
```

Comment:
here as well

Reply:
I think i did that

Reply:
argh, there is 3 examples since yesterday, not two. 
Mea culpa

Reply:
9e7ecfbdcae6365aa65339aee2316e9baac45586